code-of-conduct.md

```diff
@@ -0,0 +1,84 @@
+# Contributor Covenant Code of Conduct
```

Comment:
This one is actually taken from:
https://www.contributor-covenant.org/version/2/0/code_of_conduct/ (the license details are at the bottom)
I wasn't familiar with it, but I like it, because of how concise and clear it is.

By PSF's CoC, did you mean this one: https://www.python.org/psf/conduct/
I just glanced at it, and I think it would take some time to rewrite it (because it invokes some very PSF-specific venues / events etc.). I don't see a large substantive difference between "Our Standards" sections of the two CoCs. The largest difference is in how they go about "Enforcement". The Contributor Covenant version (the one here) is I think clearer in describing consequences of CoC violations whereas PSF's CoC does a good job at outlining the procedure of an investigation.

My proposal is to stick to the Contributor Covenant version, but if there are investigations we could draw inspiration from the PSF's procedure.

Thoughts?

Reply:
I guess my point is not to write our own CoC at all. CoCs evolve, and I think it's not something we necessarily want to maintain ourselves. We're also in the Python/PyData ecosystem, so having those same CoCs kinda makes sense.

In terms of venues and events, PSF can set the standards for the events and venues they sponsor and own, but it doesn't mean other non-PSF projects can't abide by the same rules of play.

So my suggestion would be to have a very short CoC section, which says we want people to feel safe, and now harassment is tolerated, and that we follow PSF's CoC, for details refer to that CoC.

It shortens the document substantially. I'm still not happy about how long the whole thing is (including the trademark and antitrust parts)

Reply:
I dug around a little bit and then remembered that early on we had asked a colleague who has studied GitHub communities for feedback re. our repository, and they suggested that we should adopt our own projects-specific code of conduct, and also suggested contributor covenant. (They in fact commented that project-specific CoC is better than just linking to GitHub community guidelines, as we [do in our docs](https://fairlearn.org/v0.6.0/about/index.html#code-of-conduct) currently.)

That also seemed a recommended practice [here](https://opensourcediversity.org/#basics).

So I'm starting to leani towards our own CoC. There are also some additional pragmatic reasons--we should be aware of any changes in CoC.

Reply:
If we prefer something shorter, we could possibly use [Contributor Covenant 1.4](https://www.contributor-covenant.org/version/1/4/code-of-conduct/).